src/Corax/IndexFieldsMapping.cs

```diff
@@ -1,24 +1,28 @@
 ﻿using System;
 using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Runtime.InteropServices;
 using Sparrow.Server;
+using Sparrow.Threading;
 using Voron;
 
 namespace Corax;
 
-public abstract class IndexFieldsMappingBase : IEnumerable<IndexFieldBinding>
+public abstract class IndexFieldsMappingBase : IEnumerable<IndexFieldBinding>, IDisposable
```

Comment:
How is this actually being used?

Typically we have two phases, creation (mutated) and usage (read only)(